### PR TITLE
implementation (locales): #29040 #28477 Locales feedback & preserve table sorting.

### LINKED ISSUE #28477: Add / Edit new Language
### Parent Issue

#28474

### User Story

As a user, I need the ability to add new Locale for eventual listing. This will provide me with a comprehensive view of the available languages within the system and empower me to contribute new languages as needed.

### Acceptance Criteria

- [ ] Users should be able to add new locales via a user-friendly interface integrated into the listing table.
- [ ] Follow Designs. 

### Proposed Objective

User Experience

### Proposed Priority

Priority 2 - Important

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

https://www.figma.com/file/tTWRVYunjnzTAWaj8Ctvra/Locales-%2F-Languages?node-id=472-21986&mode=dev

DEMO: https://www.figma.com/proto/tTWRVYunjnzTAWaj8Ctvra/Locales-%2F-Languages?page-id=0%3A1&type=design&node-id=356-41347&viewport=-1747%2C-4187%2C0.26&t=fx8jP5j5bAGXSHbC-1&scaling=scale-down&starting-point-node-id=356%3A41347&show-proto-sidebar=1

Reference: https://github.com/dotCMS/core/issues/25605 

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

### LINKED ISSUE #29040: Locales: Save sorting in the main table.
### User Story
Preserve the sort order while in the session in the Locale table, so when the user returns to the page, they see the data sorted in the way they defined previously.

### Acceptance Criteria

- The sort order is preserved and applied when the user returns to the page.

### dotCMS Version
master

### Proposed Objective
Enhance user experience by preserving the sort order in the Locale table, allowing users to return to their preferred view without needing to re-sort the data.

### Proposed Priority
_to_define_

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.
_to_define_

### Assumptions & Initiation Needs
_to_define_

### Quality Assurance Notes & Workarounds
_to_define_

### Sub-Tasks & Estimates

### PR DESCRIPTION
**Pull request by hmoreras**
### Proposed Changes
* Directive that preserve the order of a table when the user leave the page, even when `sortField` or `sortOrder` are defined. 
* QA feedback: regarding missing copy button tooltip. 
* QA feedback: Locale header don't show the icon when sorted.   

### Screenshots

https://github.com/dotCMS/core/assets/31667212/4ca4d80a-3f48-45bc-b724-b7607d7e4707

This PR fixes: #29040

### CODE REVIEW

File: core-web/libs/portlets/dot-locales/portlet/src/lib/dot-locales-list/components/dot-locale-create-edit/DotLocaleCreateEdit.component.html [hunk 1]
```diff
@@ -21,7 +21,10 @@
         <div class="field">
             <label for="isoCodeEdit">{{ 'locale.id' | dm }}</label>
             <span class="p-input-icon-right">
-                <i [cdkCopyToClipboard]="localeIDInput.value" class="pi pi-copy"></i>
+                <dot-copy-button
+                    [copy]="localeIDInput.value"
+                    *ngIf="localeIDInput.value"
```
Comment (nicobytes) [inline core-web/libs/portlets/dot-locales/portlet/src/lib/dot-locales-list/components/dot-locale-create-edit/DotLocaleCreateEdit.component.html:26]: @hmoreras use the new Angular syntax with @if

File: core-web/libs/portlets/dot-locales/portlet/src/lib/dot-locales-list/components/dot-locale-create-edit/DotLocaleCreateEdit.component.html [hunk 2]
```diff
@@ -34,7 +37,10 @@
         <div class="field">
             <label for="id">{{ 'locale.language.id' | dm }}</label>
             <span class="p-input-icon-right">
-                <i [cdkCopyToClipboard]="languageIDInput.value" class="pi pi-copy"></i>
+                <dot-copy-button
+                    [copy]="languageIDInput.value"
+                    *ngIf="languageIDInput.value"
```
Comment (nicobytes) [inline core-web/libs/portlets/dot-locales/portlet/src/lib/dot-locales-list/components/dot-locale-create-edit/DotLocaleCreateEdit.component.html:42]: @hmoreras use the new Angular syntax with @if

File: core-web/libs/ui/src/lib/directives/dot-state-restore/dot-state-restore.directive.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    Session = 'session',
+    Local = 'local'
+}
+
+@Directive({
+    selector: '[dotStateRestore]',
+    standalone: true
+})
+export class DotStateRestoreDirective implements AfterViewInit {
+    constructor(private table: Table) {}
```
Comment (nicobytes) [inline core-web/libs/ui/src/lib/directives/dot-state-restore/dot-state-restore.directive.ts:22]: @hmoreras Use the `inject` function instead to handle injections in the constructor.

Comment (oidacra) [reply core-web/libs/ui/src/lib/directives/dot-state-restore/dot-state-restore.directive.ts:22]: Yes, use `inject()`

**Review by oidacra — APPROVED** [review]

**Review by nicobytes — APPROVED** [review]
LGTM 🎉

File: core-web/libs/portlets/dot-locales/portlet/src/lib/dot-locales-list/dot-locales-list.component.html [hunk 4]
```diff
@@ -10,7 +10,7 @@
                 pInputText />
         </span>
         <button
-            (click)="store.openAddEditDialog()"
+            (click)="store.openAddEditDialog(null)"
```
Comment (rjvelazco) [inline core-web/libs/portlets/dot-locales/portlet/src/lib/dot-locales-list/dot-locales-list.component.html:13]: Do you know if we can pass `empty`? `tore.openAddEditDialog()`

You can also set `null` as the default value:

```js
openAddEditDialog = ( value = null ) => {
// ...
}
```

**Review by rjvelazco — APPROVED** [review]

**Pull request merged**